### ISSUE TITLE
 Different results for restore with flag recovery_target= 'immediate' for postgres <=11 and postgres >=12

### ISSUE THREAD

**Issue by woblerr** [pgbackrest/pgbackrest#1314]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: `2.30`

2. PostgreSQL version: `11 and 12`

3. Operating system/version: `Oracle Linux Server release 7.6`

4. Did you install pgBackRest from source or from a package? `package`

5. Describe the issue:

For different postgres versions we got different results after executing the same `restore` command with `--type immediate` flag.

Restore command:

```bash
pgbackrest restore --stanza some-name --type immediate --tablespace-map-all /var/lib/pgsql/tablespace --log-level-console info
```

Using `postgresql.auto.conf`:

```bash
archive_command = '/bin/true'
archive_mode = 'on'
max_wal_senders = '3'
listen_addresses = '*'
wal_level = 'hot_standby'
hot_standby = 'off'
max_connections = '500'
```

With these restore parameters, the following lines are added to `recovery.conf` or `postgresql.auto.conf`, depending on the version:

```bash
restore_command = 'pgbackrest --stanza=some-name archive-get %f "%p"'
recovery_target = 'immediate'
```

We performed a recovery for 2 different instances (11 and 12 versions) and got different results:

* for postgres version 12: the database was restored successfully, but wasn't started (wasn't promoted);
* for postgres version 11: the database was restored successfully and started (was promoted).

According to the pgbackrest documentation `11.1.11 Target Action Option (--target-action)`:

```
This option is effective when hot_standby=on is configured in postgresql.conf, otherwise the cluster will be promoted when the target is reached or there is no more WAL in the archive.
```

The reason is that the file `recovery.signal` is created for postgres >=12. In our case, `hot_standby ='off'` and `recovery_target_action` parameter isn't set, so the `shutdown` value is used. For this case according postgres documentation, any subsequent startup will end with an immediate shutdown, unless the configuration is changed or the recovery.signal file is manually deleted.

However, going back to pgbackrest, we didn't expect to get different behavior for the same recovery settings. It is clear that these are features of different postgres versions and this problem isn't directly relates to pgbackrest, but it may be worth specifying this feature in documentation. The reasons for the different results for the same restore command are not so obvious.

For information, piece postgres startup logs:
<details>
<summary> postgres 11:</summary>

    LOG:  starting point-in-time recovery to earliest consistent point
    LOG:  restored log file "00000001000000030000000D" from archive
    LOG:  redo starts at 3/D000028
    LOG:  consistent recovery state reached at 3/D00E178
    LOG:  recovery stopping after reaching consistency
    LOG:  redo done at 3/D00E178
    LOG:  last completed transaction was at log time 2021-02-26 01:01:35.715376
    LOG:  selected new timeline ID: 2
    LOG:  archive recovery complete
    LOG:  database system is ready to accept connections
    LOG:  redo starts at 3/D00E1E8
    LOG:  invalid record length at 3/D00E220: wanted 24, got 0
    LOG:  redo done at 3/D00E1E8
    LOG:  database system is ready to accept connections

</details>

<details>
<summary> postgres 12:</summary>

    LOG:  starting point-in-time recovery to earliest consistent point
    LOG:  restored log file "000000090000009300000025" from archive
    LOG:  redo starts at 93/25000028
    LOG:  consistent recovery state reached at 93/250122F0
    LOG:  recovery stopping after reaching consistency
    LOG:  shutdown at recovery target
    LOG:  shutting down
    LOG:  database system is shut down

</details>

<details>
<summary> postgres 12 with `recovery_target_action = 'promote'` :</summary>

    LOG:  starting point-in-time recovery to earliest consistent point
    LOG:  restored log file "000000090000009300000025" from archive
    LOG:  redo starts at 93/25000028
    LOG:  consistent recovery state reached at 93/250122F0
    LOG:  recovery stopping after reaching consistency
    LOG:  redo done at 93/250122F0
    LOG:  last completed transaction was at log time 2021-02-26 01:01:09.449048
    LOG:  selected new timeline ID: 10
    LOG:  archive recovery complete
    LOG:  database system is ready to accept connections
    LOG:  redo starts at 93/25012368
    LOG:  invalid record length at 93/250123A0: wanted 24, got 0
    LOG:  redo done at 93/25012368
    LOG:  database system is ready to accept connections

</details>

**Comment by dwsteele** [comment]
This looks related to #987, @cbandy.

Honestly we just haven't gotten around to updating the docs. If someone could test the configurations and submit a PR with updates to the docs here (https://github.com/pgbackrest/pgbackrest/blob/master/doc/xml/reference.xml#L1530) that would be really helpful.

No need to get the formatting right. Just get some text into the XML file and we'll take care of formatting.

**Comment by cbandy** [comment]
The ask here seems to be that pgBackRest document how Postgres behaves when omitting the `--target-action` option.

I'm not sure where to capture that, and I cannot tell if this would be fixed by #1960.

**Comment by dwsteele** [comment]
I think this is fixed by PR #1960. Thoughts, @woblerr?

**Comment by woblerr** [comment]
This looks good to me, thanks.
